### DISCUSSION TITLE
`/boot/extlinux/extlinux.conf`  configuration FDT does not work.

### DISCUSSION THREAD

**Discussion by leadercxn** [OE4T/meta-tegra#1180] [General]
Hi OE4T developer,

I add 
```shell
UBOOT_EXTLINUX_FDTDIR = "/boot"
UBOOT_EXTLINUX_FDT = "/boot/${DTBFILE}"

cboot-extlinux 
```
in my yocto local.conf and I can see /boot/extlinux/extlinux.conf  in my system.

Now ,I generate a .dtbo  and use `fdtoverlay`  to merge .dtb and .dtbo, and then use `fdtdump` to check the .dtb files are merged successfully, replace the original dtb file with the merged dtb file in `/boot`. After the reboot, use `dtc -I fs /proc/device-tree > dts.txt` to find that the merged dtb was not working.  Is there something wrong with my operation? Or am I missing something?

<img width="1221" alt="image" src="https://user-images.githubusercontent.com/43691182/220281871-7d9a101b-08dc-4e9d-9b81-ad0c76eeecc0.png">

**Comment by madisongh** [comment]
Take a look at the messages generated by cboot on the serial console.  Do you see logs from it about processing the extlinux.conf file, loading the kernel DTB, etc.?  Are there any error messages in that sequence?

**Reply by leadercxn** [reply]
The entire boot log, I just found a print of `extlinux` here
![image](https://user-images.githubusercontent.com/43691182/220374360-49f10d2d-0335-4dbd-a32d-148b86df02ab.png)

**Reply by leadercxn** [reply]
@madisongh  And if I modify the `FDT` in`extlinux.conf` file, the name of the device tree `tegra194-p3668-all-p3509-0000-user.dtb` was not found in the entire startup log, so I guess the system did not reference it?
<img width="652" alt="image" src="https://user-images.githubusercontent.com/43691182/220506435-bf4a1dfa-d75d-4916-908f-c336ad9aa83b.png">

**Reply by leadercxn** [reply]
[xavier-nx-yocto.log](https://github.com/OE4T/meta-tegra/files/10800779/xavier-nx-yocto.log)
All logs output from the serial port

**Reply by leadercxn** [reply]
I think yocto project may still have some problems with its cboot,  I used the cboot_t194.bin file released by NVIDIA [r32.7.1](https://developer.nvidia.com/embedded/linux-tegra-r3271) to replace the one generated by yocto, it works.
<img width="1088" alt="image" src="https://user-images.githubusercontent.com/43691182/220578136-dd7a0deb-f27d-44bb-b424-b615d42797e9.png">
The attachment is the startup log output from the serial port
<img width="824" alt="image" src="https://user-images.githubusercontent.com/43691182/220578290-39f2ab8b-903d-4a2f-86ea-a75a93467610.png">
[xavier-nx-yocto.log](https://github.com/OE4T/meta-tegra/files/10802041/xavier-nx-yocto.log)
BTW, I use branch `kirkstone-l4t-r32.7.x`

**Reply by madisongh** [reply]
Yeah, it sounds like the patches we have to fix other issues in NVIDIA's cboot ext4 filesystem support need to be looked at again.

You can, in the meantime, configure `PREFERRED_PROVIDER_virtual/bootloader` and `PREFERRED_PROVIDER_cboot` to be `"cboot-prebuilt"` to use the pre-built binary instead of building from sources.

**Reply by madisongh** [reply]
Also, can you try this with the latest on the branch?  The error message mentioned in the screen cap of the failure case above isn't in our patches any longer, when we updated from L4T R32.7.2 to L4T R32.7.3.

**Reply by leadercxn** [reply]
> Also, can you try this with the latest on the branch? The error message mentioned in the screen cap of the failure case above isn't in our patches any longer, when we updated from L4T R32.7.2 to L4T R32.7.3.

Okay, I compiled my project using this branch` kirkstone-l4t-r32.7.x` of that repository `tegra-demo-distro`, and the resulting that 
 `meta-tegra` commit is too old.

**Reply by leadercxn** [reply]
> Also, can you try this with the latest on the branch? The error message mentioned in the screen cap of the failure case above isn't in our patches any longer, when we updated from L4T R32.7.2 to L4T R32.7.3.

Greate, that works~ thanks
